app/api/coreApi.js

```diff
@@ -905,6 +913,82 @@ function getBlockByHashWithTransactions(blockHash, txLimit, txOffset) {
 	});
 }
 
+const getBlockListDefaultArgs = {
+	limit: config.site.browseBlocksPageSize,
+	offset: 0,
+	sort: 'desc'
+};
+
+function getBlockList(args)
+{
+	args = Object.assign(Object.assign({}, getBlockListDefaultArgs), args);
+
+	return new Promise(function(resolve, reject) {
+		getBlockchainInfo().then(function(getblockchaininfo) {
+			var sortDesc = args.sort == 'desc';
+
+			// Get all the block heights we will display
+			var blockHeights = Array.from({length: args.limit})
+				.map((_, i) => sortDesc ? (getblockchaininfo.blocks - args.offset - i) : (args.offset + i))
+				.filter(h => h >= 0 && h <= getblockchaininfo.blocks);
+
+			// hack: default regtest node returns getblockchaininfo.blocks=0, despite having a genesis block
```

Comment:
This needs some testing for regtest, not sure if it is correct like this.

Reply:
this is kind of "hack" has been used elsewhere in the code to deal with `regtest`, so I'm good with it